Lib/unittest/test/testmock/testhelpers.py

```diff
@@ -871,6 +872,19 @@ def test_autospec_on_bound_builtin_function(self):
         mocked.assert_called_once_with(4, 5, 6)
 
 
+    def test_autospec_getattr_partial_function(self):
+        # bpo-32153 : getattr returning partial functions without
+        # __name__ should not create AttributeError in create_autospec
+        class Foo(object):
+
+            def __getattr__(self, name):
+                return partial(lambda name: name, name)
+
+        proxy = Foo()
+        autospec = create_autospec(proxy)
+        assert not hasattr(autospec, '__name__')
```

Comment:
Why not use self.assertFalse here?

Reply:
Changed to use `self.assertFalse`. Thanks.